src/client/mercury/mercury.js

```diff
@@ -166,9 +166,14 @@ var Mercury = SparkBase.extend(
   }),
 
   _attemptConnection: function _attemptConnection() {
+    var socket = this._getNewSocket();
```

Comment:
Couldn't this just be `var socket = new Socket()`? or does `_getNewSocket()` make tests easier?